### PR TITLE
Add tools plugin with import struct feature

### PR DESCRIPTION
**Pull request by DimalChandrasiri**

### CODE REVIEW

File: composer/modules/web/src/plugins/import-struct/commands.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (srsharon) [inline composer/modules/web/src/plugins/import-struct/commands.js]: License year need to be updated

Comment (keizer619) [inline composer/modules/web/src/plugins/import-struct/commands.js]: This has to be 2018

File: composer/modules/web/src/plugins/import-struct/handlers.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (srsharon) [inline composer/modules/web/src/plugins/import-struct/handlers.js]: License year need to be updated

Comment (keizer619) [inline composer/modules/web/src/plugins/import-struct/handlers.js]: change it to 2018

File: composer/modules/web/src/plugins/import-struct/menus.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (srsharon) [inline composer/modules/web/src/plugins/import-struct/menus.js]: License year need to be updated

File: composer/modules/web/src/plugins/tools/constants.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (srsharon) [inline composer/modules/web/src/plugins/tools/constants.js]: License year need to be updated

File: composer/modules/web/src/plugins/tools/menus.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (srsharon) [inline composer/modules/web/src/plugins/tools/menus.js]: License year need to be updated

File: composer/modules/web/src/plugins/tools/plugin.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (srsharon) [inline composer/modules/web/src/plugins/tools/plugin.js]: License year need to be updated

File: composer/modules/web/src/plugins/tools/constants.js (hunk 4 above)
Comment (DimalChandrasiri) [reply composer/modules/web/src/plugins/tools/constants.js]: updated with commit f29a50ca6573aae3b2ac8e0681ec0d8c47144cfb

File: composer/modules/web/src/plugins/tools/menus.js (hunk 5 above)
Comment (DimalChandrasiri) [reply composer/modules/web/src/plugins/tools/menus.js]: updated with commit f29a50ca6573aae3b2ac8e0681ec0d8c47144cfb

File: composer/modules/web/src/plugins/tools/plugin.js (hunk 6 above)
Comment (DimalChandrasiri) [reply composer/modules/web/src/plugins/tools/plugin.js]: updated with commit f29a50ca6573aae3b2ac8e0681ec0d8c47144cfb

File: composer/modules/web/src/plugins/import-struct/handlers.js [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+ * @returns {Object[]} command handler definitions.
+ *
+ */
+export function getHandlerDefinitions(plugin) {
+    return [
+        {
+            cmdID: COMMANDS.SHOW_IMPORT_STRUCT_DIALOG,
+            handler: () => {
+                const topLevelNodes = plugin.appContext.editor.getActiveEditor().getProperty('ast');
+                let structNode = DefaultNodeFactory.createStruct();
```
Comment (keizer619) [inline composer/modules/web/src/plugins/import-struct/handlers.js]: @DimalChandrasiri ; can't we make this "const" as its not initialized?

File: composer/modules/web/src/plugins/import-struct/dialogs/import-struct-dialog.jsx [hunk 8]
```diff
@@ -56,7 +56,8 @@ class ImportStructDialog extends React.Component {
      * Called when user clicks 'Import Struct' menu item.
      */
     onImportJson() {
-        if (this.props.onImport(this.state.json)) {
+        const { structName } = this.state;
```
Comment (keizer619) [inline composer/modules/web/src/plugins/import-struct/dialogs/import-struct-dialog.jsx]: @DimalChandrasiri You can directly pass this.state.structName as its been done for this.state.json. I think better to follow one approach for both params

Comment (DimalChandrasiri) [reply composer/modules/web/src/plugins/import-struct/dialogs/import-struct-dialog.jsx]: fixed with commit 0974ece26d1cb2557d0f080e28c729a8122f138d

File: composer/modules/web/src/plugins/import-struct/handlers.js (hunk 7 above)
Comment (DimalChandrasiri) [reply composer/modules/web/src/plugins/import-struct/handlers.js]: fixed with commit 0974ece26d1cb2557d0f080e28c729a8122f138d

**Pull request merged**